### DISCUSSION TITLE
use xformers on linux with diffusers==0.13.0

### DISCUSSION THREAD

**Discussion by chengziyi** [kohya-ss/sd-scripts#210] [General]
when i run this code on linux with xformers
it reports ERROR: "forward_xformers() got an unexpected keyword argument 'encoder_hidden_states'"

i found a way to avoid it, not the elegant way, but can work with diffuser==0.13.0
python==3.9 , torch==1.13.1, xformers==0.0.16rc425

1. find function: replace_unet_cross_attn_to_xformers()  in library/train_util.py 
remove this line: "diffusers.models.attention.CrossAttention.forward = forward_xformers"

2.open site-packages/diffusers/models/attention.py, find "class BasicTransformerBlock", add "self.attn1.set_use_memory_efficient_attention_xformers(True)" after self.attn1 = CrossAttention(***)

**Comment by CHR-ray** [comment]
Thank you, I have reinstall several version of xformers, torch, cuda... 
None of them really work, only yours.
My setting:
python 3.10
torch                     1.13.1
torchvision               0.14.1
diffusers                 0.13.1 
xformers                  0.0.16

**Reply by CHR-ray** [reply]
sorry, I reply too early. When I restart my lora script, it even raise out of vram at 1024 * 1024.... (24GB VRAM I have)

Maybe there are still something wrong...

**Comment by chengziyi** [comment]
PS: what i do is avoid this ERROR: "forward_xformers() got an unexpected keyword argument 'encoder_hidden_states'" with diffusers==0.13 on linux.  Obviously it is not a solution to solve the problem
